### ISSUE TITLE
[animation] transition in foreach: wrong order

### ISSUE THREAD

**Issue by seb-odoo** [odoo/owl#655]
I was investigating another issue with animations (still not reproducible) and I stumbled upon this one.

Run the following on the playground.

Click on "id: 2"
Notice how "id: 2" is fading out between "0" and "1"

It should be expected that "id: 2" is fading out between "1" and "3" instead.

```js
// The goal of this component is to see how the t-transition directive can be
// used to generate simple transition effects.
const { Component, useState } = owl;

class Counter extends Component {}

class App extends Component {
    constructor() {
        super(...arguments);
        this.state = useState({ id: 0 });
    }
}
App.components = { Counter };

const app = new App();
app.mount(document.body);
```
```xml
<templates>
  <button t-name="Counter">
    id: <t t-esc="props.id"/>
  </button>

  <div t-name="App">
    <t t-foreach="[0, 1, 2, 3, 4]" t-as="n" t-key="n">
      <Counter t-if="state.id !== n" id="n" t-transition="fade" t-on-click="state.id = n"/>
    </t>
  </div>
</templates>
```
```css
.fade-leave-active {
    transition: color 2s;
}
.fade-leave-to {
    color: red;
}
```

**Comment by seb-odoo** [comment]
As for the other issue, I can't reproduce it outside of my code, but here is how it goes:

I have a list of messages. The list is a component and the messages too. The messages are displayed with a `foreach`. When a message is removed from the list, we want an animation.

For this, I have added a `t-transition` on the message, on the xml of the list. (note: I would like to add the transition attribute directly on the message xml instead, but it doesn't work either)

I have added adequate css.

Here is the technical flow:
- `transitionRemove` is called,  `-leave` and `-leave-active` are properly added
- `nextFrame` & `whenTransitionEnd` are called, `transitionend` is bound
And here is the issue, at this point:
- `transitionend` is never triggered
- the dom does not contain any of the `leave` classes anymore

I suspect this could be due do a render removing them, before the animation ends.
